document/src/vespa/document/config/doctypecfg.h

```diff
@@ -0,0 +1,16 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+/**
+ * Forward-declaration of the types
+ * DocumenttypesConfig and DocumenttypesConfigBuilder
+ * (globally visible).
+ **/
+
+namespace document::config::internal {
+class InternalDocumenttypesType;
+}
+
+using DocumenttypesConfigBuilder = document::config::internal::InternalDocumenttypesType;
+using DocumenttypesConfig = const document::config::internal::InternalDocumenttypesType;
```

Comment:
Consider moving the using statements to the `document::config` namespace, i.e. the same namespace that is used when the generated `config-documenttypes.h` is included.

Consider renaming the header file to something containing fwd, e.g. as `iosfwd` contains forward declarations for `iostream`.